api/v1beta1/azuremachinetemplate_webhook.go

```diff
@@ -78,35 +92,42 @@ func (r *AzureMachineTemplate) ValidateUpdate(oldRaw runtime.Object) error {
 
 		// We need to set ssh key explicitly, otherwise Default() will create a new one.
 		if old.Spec.Template.Spec.SSHPublicKey == "" {
-			old.Spec.Template.Spec.SSHPublicKey = r.Spec.Template.Spec.SSHPublicKey
+			old.Spec.Template.Spec.SSHPublicKey = t.Spec.Template.Spec.SSHPublicKey
 		}
 
-		old.Default()
+		err := r.Default(ctx, old)
+		if err != nil {
+			allErrs = append(allErrs,
```

Comment:
We're adding a new error to `allErrs` here but not wrapping or otherwise preserving the initial `err`. Should we?

Reply:
done